File: src/com/kedzie/vbox/api/IMediumFormat.java
package com.kedzie.vbox.api;

import java.util.List;
import java.util.Map;

import com.kedzie.vbox.soap.KSOAP;

@KSOAP
public interface IMediumFormat extends IManagedObjectRef {

    @KSOAP(cacheable=true) public String getId();
    @KSOAP(cacheable=true) public String getName();
    @KSOAP(cacheable=true) public Integer getCapabilities();
    @KSOAP(cacheable=true) public Map<String, List<String>> describeFileExtensions();
    @KSOAP(cacheable=true) public Map<String, List<String>> describeProperties();
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IMediumFormat.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

@KsoapProxy
@Ksoap
interface IMediumFormat : IManagedObjectRef, Parcelable {

    @Cacheable("Id")
	suspend fun getId(): String

    @Cacheable("Name")
	suspend fun getName(): String

    @Cacheable("Capabilities")
	suspend fun getCapabilities(): Int

    @Cacheable
    suspend fun describeFileExtensions(): Map<String, List<String>>

    @Cacheable
    suspend fun describeProperties():Map<String, List<String>>
}